components/parsehub/parsehub.app.mjs

```diff
@@ -1,11 +1,91 @@
+import { axios } from "@pipedream/platform";
+
 export default {
   type: "app",
   app: "parsehub",
-  propDefinitions: {},
+  propDefinitions: {
+    projectName: {
+      type: "string",
+      label: "Project Name",
+      description: "The name of the project to trigger a new run for",
+      required: true,
+    },
+    projectId: {
+      type: "string",
+      label: "Project ID",
+      description: "The unique identifier of the project",
+      async options() {
+        const response = await this.listProjects({});
+        const projectsIDs = response.projects;
+        return projectsIDs.map(({
+          token, title,
+        }) => ({
+          value: token,
+          label: title,
+        }));
+      },
+    },
+    runToken: {
+      type: "string",
+      label: "Run Token",
+      description: "The token of the run to retrieve data for",
+    },
+    startUrl: {
+      type: "string",
+      label: "Start URL",
+      description: "The url to start running on",
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _baseUrl() {
+      return "https://www.parsehub.com/api/v2";
+    },
+    async _makeRequest(opts = {}) {
+      const {
+        $ = this,
+        path,
+        params,
+        ...otherOpts
+      } = opts;
+      return axios($, {
+        ...otherOpts,
+        url: this._baseUrl() + path,
+        params: {
+          ...params,
+          api_key: `${this.$auth.api_key}`,
+        },
+      });
+    },
+    async getProjectDetails({
```

Comment:
Optional: The `async` isn't really needed here. Applies to all the app file methods.
```suggestion
    getProjectDetails({
```